core/src/impl/Kokkos_Half_FloatingPointWrapper.hpp

```diff
@@ -215,10 +216,47 @@ cast_from_wrapper(const Kokkos::Experimental::bhalf_t& x);
 /************************** END forward declarations **************************/
 
 namespace Impl {
+
+template <typename FloatType>
+struct BitComparisonWrapper {
+  std::uint16_t value;
+};
+
+template <typename FloatType>
+static constexpr BitComparisonWrapper<FloatType> exponent_mask;
```

Comment:
Declare both masks next to each other and specialize them both in one shot, it means less macro guards and will improve readability.